File: unit-1.1-abstract_data_types/1.1.4-practices/interface-vs-implementation.cpp
#include <iostream>
#include <string>
using namespace std;

class MyClass {
	private:
	  string first, second;

	public:
	  void set(string, string);
	  string getFirst();
	  string getSecond();
}

void MyClass::set(string x, string y) {
  first = x;
  second = y;
}

string MyClass::getFirst() { return first; }

string MyClass::getSecond() { return second; }

int main() {

  MyClass objectA;

  objectA.set("Hello, ", "World!");

  cout << objectA.getFirst() << objectA.getSecond() << endl;

  return 0;
}

File: unit-1.1-abstract_data_types/1.1.4-practices/interface-vs-implementation.rs
struct MyClass {
  first: String,
  second: String,
}

impl MyClass {
  fn set(&mut self, f: String, s: String) {
    self.first = f;
    self.second = s;
  }

  fn get_first(&self) -> &str {
    &self.first
  }

  fn get_second(&self) -> &str {
    &self.second
  }
}

fn main() {
  let mut object_a = MyClass {
    first: String::new(),
    second: String::new(),
  };

  object_a.set(String::from("Hello, "), String::from("World!"));

  println!("{}{}", object_a.get_first(), object_a.get_second());
}
